src/components/SliderField/SliderField.tsx

```diff
@@ -0,0 +1,204 @@
+import * as React from 'react'
+import './SliderField.css'
+
+export type SliderFieldProps = {
+  header: string
+  className?: string
+  min?: number
+  max?: number
+  defaultValue?: number | readonly [number, number]
+  label?: string | React.PureComponent<{ value: number | [number, number] }>
+  range?: boolean
+  onChange?: (
+    ev: React.ChangeEvent<HTMLInputElement>,
+    data: number | [number, number]
+  ) => void
+  onMouseUp?: (
+    ev: React.MouseEvent<HTMLInputElement>,
+    data: number | [number, number]
+  ) => void
+}
+
+export enum SliderLastInteraction {
+  'from',
+  'to'
+}
+export enum SliderDefault {
+  MIN = 0,
+  MAX = 100,
+  FROM = 0,
+  TO = 100
+}
+
+export type SliderFieldState = {
+  from: number
+  to: number
+  lastInteraction: SliderLastInteraction
+}
+
+export class SliderField extends React.PureComponent<
+  SliderFieldProps,
+  SliderFieldState
+> {
+  state = {
+    from: SliderDefault.FROM,
+    to: SliderDefault.TO,
+    lastInteraction: SliderLastInteraction.from
+  }
+
+  componentDidMount(): void {
+    if (this.props.defaultValue === undefined) {
+      this.setState((prevState) => {
+        return {
+          ...prevState,
+          from: !isNaN(this.props.min) ? this.props.min : prevState.from,
+          to: !isNaN(this.props.max) ? this.props.max : prevState.to
+        }
+      })
+    } else {
+      if (Array.isArray(this.props.defaultValue)) {
+        this.setState({
+          from: this.props.defaultValue[0],
+          to: this.props.defaultValue[1]
+        })
+      } else {
+        this.setState({
+          to: Number(this.props.defaultValue)
+        })
+      }
+    }
+  }
```

Comment:
To keep consistent with the rest of the code, what do you think of destructuring props here? like this:

```ts
  componentDidMount(): void {
    const { defaultValue, min, max } = this.props

    if (defaultValue === undefined) {
      this.setState((prevState) => {
        return {
          ...prevState,
          from: !isNaN(min) ? min : prevState.from,
          to: !isNaN(max) ? max : prevState.to
        }
      })
    } else {
      if (Array.isArray(defaultValue)) {
        this.setState({
          from: defaultValue[0],
          to: defaultValue[1]
        })
      } else {
        this.setState({
          to: Number(defaultValue)
        })
      }
    }
  }
```

(coded here, beware ⚠️ )

Reply:
done